tutorials/haskell/snap-web-app/tutorial.md

```diff
@@ -0,0 +1,490 @@
+---
+title: How to create a web application in Haskell with Snap
+published: 2017-01-30
+ghc: 8.0.1
+lts: 7.18
+libraries: snap-core-1.0.1.0 snap-server-1.0.1.1 heist-1.0.1.0 HDBC-sqlite3-2.3.3.1    
+language: haskell
+description: In this tutorial we show you how to build a simple TODO web app using Snap framework
+author-name: Omar García
+---
+
+## Introduction
+
+[Snap](http://snapframework.com/) is a simple and [fast](http://snapframework.com/blog/2010/11/17/snap-0.3-benchmarks)
+web development framework for the Haskell programming language with a clean
+[monads](https://wiki.haskell.org/Monad) oriented design. Snap apps are often
+composed by [snaplets](http://snapframework.com/docs/tutorials/snaplets-tutorial),
+auto-contained pieces of reusable functionalities that can be glued together to build a website.
+
+In this tutorial we'll show you how to build a simple TODO web app using Snap,
+[SQLite](https://www.sqlite.org/), and [Stack](http://docs.haskellstack.org/en/stable/README/)
+as our Haskell build tool. The target app should deliver the following features:
+
+ - List Todo items
+ - Create a new Todo item
+ - Mark a Todo item as Done
+ - Delete a Todo item
+
+
+## Getting Started
+
+Snap comes shipped with a [scaffolding utility](http://snapframework.com/docs/quickstart)
+to scrap some boilerplate. So, when you first run the `snap init` command inside an empty
+directory, it will create a bunch of files for you. But let's leave that aside,
+for now let's focus on how to create a Snap web app from scratch using Stack.
+
+Firstly, let's [create a project using Stack](http://docs.haskellstack.org/en/stable/README/#start-your-new-project) by running the following command:
+
+```
+$ stack new todo-webapp
+```
+
```

Comment:
I think that if this tutorial is aimed at Haskell beginners, we need to add a couple of sentences here about what `stack new` is doing:
-creating the directory containing our project, which is named `todo-webapp` (because that's what we named it)
-creating the src and test directories in our project directory
-creating the cabal file
-we should point out that the cabal file is where stack looks to see which dependencies it needs to fetch for us.